### PR TITLE
feat: [CODE-713]: Expose Gitness API service to harness-core-ui

### PR DESCRIPTION
**Pull request by tan-nhu**
### Summary

Expose Gitness API service to harness-core-ui. The service is required to implement Gitness/CI integration.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by tan-nhu** [comment]
retrigger titlecheck

**Review by shaurya-harness — DISMISSED** [review]

File: configs/restful-react.config.js [hunk 1]
```diff
@@ -240,6 +240,15 @@ module.exports = {
       base: `{getConfig("ccm/public-pricing/api/v1")}`
     }
   },
+  code: {
```
Comment (abhinavrastogi-harness) [inline configs/restful-react.config.js:243]: can we use the new openapi client instead?
once we add this, it'll take many quarters to get it removed again due to code dependencies.

File: src/services/code/overrides.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+allowpaths: ['*']
```
Comment (abhinavrastogi-harness) [inline src/services/code/overrides.yaml:1]: why are all paths being added? we should allow only the used paths to optimise what we ship.

File: src/services/code/swagger.yaml
Comment (abhinavrastogi-harness) [inline src/services/code/swagger.yaml:1]: can we avoid committing the api spec? it bloats the repo unnecessarily.

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: configs/restful-react.config.js (hunk 1 above)
Comment (tan-nhu) [reply configs/restful-react.config.js:243]: We are using restful-react in Gitness, eventually we'll migrate to the new openapi at some point, but it needs to come from Gitness first. Right now, I don't want to make it complicated for the teams to work on with two sets of API calls.

File: src/services/code/overrides.yaml (hunk 2 above)
Comment (tan-nhu) [reply src/services/code/overrides.yaml:1]: Added allowPaths to `repos` only for now.

File: src/services/code/swagger.yaml
Comment (tan-nhu) [reply src/services/code/swagger.yaml:1]: I don't think we should remove the spec. It's there to keep track of changes from API side.

Comment (abhinavrastogi-harness) [reply src/services/code/swagger.yaml:1]: APi changes will reflect in the generated service automatically. 
You can add it to `.gitignore` if you still want to keep it in the repo.
Do you see any specific benefits from committing the API spec?

File: configs/restful-react.config.js (hunk 1 above)
Comment (abhinavrastogi-harness) [reply configs/restful-react.config.js:243]: It'll still be the same API even if it's a different client. I don't think we are planning to pass these hooks down to the microfrontend, right? Gitness MFE  already has it's own clients generated.

When is Gitness planned to be migrated to OpenAPI?
Why is this service needed in NGUI in the first place?

File: src/services/code/swagger.yaml
Comment (tan-nhu) [reply src/services/code/swagger.yaml:1]: The spec is used as a reference what is changed from backend side (the source), not what is generated (the output). It's used to communicate as the source of truth from backend. Let's say if backend publishes some breaking changes, without the source, you can't tell and track back. 

Versioning is important.

File: configs/restful-react.config.js (hunk 1 above)
Comment (tan-nhu) [reply configs/restful-react.config.js:243]: It's not the same API if you have naming different.

`When is Gitness planned to be migrated to OpenAPI?`

No date specified. I don't think Gitness is any different than other API services currently under resful-react.

`Why is this service needed in NGUI in the first place?`

It's for CIE integration. CIE needs a way to call Gitness APIs directly.

**Comment by hitesharinga** [comment]
Approving since CI team is waiting for this.

**Pull request merged**